sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2562,37 +2563,63 @@ object TimeWindowing extends Rule[LogicalPlan] {
           case a: Attribute => a.metadata
           case _ => Metadata.empty
         }
-        val windowAttr =
-          AttributeReference("window", window.dataType, metadata = metadata)()
 
-        val maxNumOverlapping = math.ceil(window.windowDuration * 1.0 / window.slideDuration).toInt
-        val windows = Seq.tabulate(maxNumOverlapping + 1) { i =>
-          val windowId = Ceil((PreciseTimestamp(window.timeColumn) - window.startTime) /
-            window.slideDuration)
+        def getWindow(i: Int, maxNumOverlapping: Int): Expression = {
```

Comment:
I'm not sure I understand `maxNumOverlapping` as a name that we tabulate over.  Isn't it like the `overlapNumber` or something?